src/lobsterpy/plotting/__init__.py

```diff
@@ -1234,3 +1235,128 @@ def get_plot(
+        self.are_coops = are_coops
+        self.are_cobis = are_cobis
+        self._bwdfs = {}  # type: ignore
+
+    @staticmethod
+    def _bwdf_data_to_plot(raw_bwdf: dict):
+        """
+        Restructure bond weighted distribution function (BWDF) data for plotting.
+
+        :param raw_bwdf: dict containing BWDF data as obtained from
+        FeaturizeIcoxxlist.calc_bwdf() or FeaturizeIcoxxlist.calc_site_bwdf()
+
+        """
+        # Keep only the BWDF data necessary for plotting
+        excluded_keys = {"centers", "edges", "bin_width", "wasserstein_dist_to_rdf"}
+
+        formatted_bwdf = {}
+        for pair, value in raw_bwdf.items():
+            if pair not in excluded_keys:
+                formatted_bwdf[pair] = {"centers": raw_bwdf.get("centers"), "icoxx_binned": value.get("icoxx_binned")}
+
+        return formatted_bwdf
+
+    def add_bwdf(self, label: str, bwdf: dict):
+        """
+        Add bond weighted distribution function (BWDF) for plotting.
+
+        :param label: Label for the BWDF. Must be unique.
+        :param bwdf: dict containing BWDF data as obtained from
+            FeaturizeIcoxxlist.calc_bwdf() or FeaturizeIcoxxlist.calc_site_bwdf()
+
+        """
+        self._bwdfs[label] = self._bwdf_data_to_plot(bwdf)
+
+    def get_plot(
+        self,
+        ax: mpl.axes.Axes | None = None,
+        colors: list[str] | None = None,
+        plot_negative: bool = True,
+        sigma: float | None = None,
+        xlim: tuple[float, float] | None = None,
+        ylim: tuple[float, float] | None = None,
+    ):
+        """
+        Get a matplotlib plot showing the bond weighted distribution function (BWDF).
+
+        :param ax: Existing Matplotlib Axes object to plot to.
+        :param colors: list of hex color codes to be used in plot
+        :param plot_negative: Will plot -1*ICOHPs. Works only for ICOHPs
+        :param sigma: Standard deviation of Gaussian broadening applied to
+            distribution data. If ,None no broadening will be
+            added.
+        :param xlim: Specifies the x-axis limits. Defaults to None for
+            automatic determination.
+        :param ylim: Specifies the y-axis limits. Defaults to None for
+            automatic determination.
+
+        Returns:
+            A matplotlib object.
+        """
+        if self.are_coops and not self.are_cobis:
+            cohp_label = "ICOOP"
+        elif self.are_cobis and not self.are_coops:
+            cohp_label = "ICOBI"
+        elif self.are_cobis and self.are_coops:
+            raise ValueError("Plot data should not contain ICOBI and ICOOP data at same time")
+        else:
+            cohp_label = "ICOHP (eV)"
+
+        if ax is None:
+            _, ax = plt.subplots()
+
+        if xlim:
+            ax.set_xlim(xlim)
+        if ylim:
+            ax.set_ylim(ylim)
+
+        palette = InteractiveCohpPlotter.COLOR_PALETTE if colors is None else colors
```

Comment:
if you use the palette from `InteractiveCohpPlotter`, remove the palette from the classes `BWDFPlotter` and `IcohpDistancePlotter`

Reply:
Good catch, fixed it to use palette from respective class